src/server.rs

```diff
@@ -410,6 +424,23 @@ impl RollupBoostServer {
                 .await
                 .unwrap_or(payload_id);
```

Comment:
It could be simpler if we removed this and just stop if the payload_id is not on the context.
```suggestion
                .unwrap_or(payload_id);
```